Automattic/wp-calypso
# Purchase Management: add new cancelation modal

### Pemise
The PR https://github.com/Automattic/wp-calypso/pull/67090 is corrupted, I opened this new and addressed the change requests.

P2: pebzTe-6r-p2

* Add new remove-plan-dialog component to the remove purchases page

#### Issue
1039-gh-Automattic/martech

#### Acceptance  Criteria
1039-gh-Automattic/martech

#### Proposed Changes
* Add a new dialog component to show up before the user starts the cancelation flow.

#### Testing Instructions
* Checkout update/pre-cancelation-modal
* Test Calypso locally
* Visit http://calypso.localhost:3000/
* Select a site with an active subscription plan
* Click on te sidebar item Upgrade / Purchases
* Select the plan
* Click on the CTA `Remove <Plan name> plan
* Verify the screenshot image is not blurry

Edge cases:
- Site isComingSoon or private: the modal won't show the site screenshot
- Mobile screen size: the modal won't show the site screenshot
- If the user are cancelling a Jetpack, Professional emails plan: the modal won't show up
- If the user are cancelling a subscription (auto renew on) within the refund window, with a domain, the modal informs them they are going to love the domain and the traffic will be redirected to the related WPCom URL.


#### Before
![before](https://user-images.githubusercontent.com/5706607/188669017-fdcb0005-8683-40db-8255-f8c9bfa970b8.gif)

#### After
https://user-images.githubusercontent.com/5706607/188671098-defa794a-7b57-435c-839c-1d31c1eef619.mp4


#### Issue
Related to 1039-gh-Automattic/martech
